addons/sale_timesheet/models/project.py

```diff
@@ -136,6 +137,13 @@ def _compute_project_overview(self):
         for project in self.filtered(lambda p: not p.project_overview):
             project.project_overview = project.allow_billable or project.allow_timesheets
 
+    @api.depends('sale_line_employee_ids', 'sale_line_employee_ids.sale_line_id', 'sale_line_id')
+    def _compute_partner_id(self):
+        for project in self.filtered(lambda p: p.allow_billable and p.allow_timesheets and p.pricing_type != 'task_rate' and not p.partner_id):
+            sol = project.sale_line_id or fields.first(project.sale_line_employee_ids.mapped('sale_line_id'))
+            if sol:
+                project.partner_id = sol.order_partner_id
```

Comment:
When you depend on "X.Y.Z", you automatically depend on "X" and "X.Y".
Also, readability matters, and `for project in self.filtered(...)` is almost always unreadable.
Also, `mapped()` is useless on relational fields.
```python
@api.depends('sale_line_employee_ids.sale_line_id', 'sale_line_id')
def _compute_partner_id(self):
    for project in self:
        if project.partner_id:
            continue
        if project.allow_billable and project.allow_timesheets and project.pricing_type != 'task_rate':
            sol = project.sale_line_id or project.sale_line_employee_ids.sale_line_id[:1]
            project.partner_id = sol.order_partner_id
```

Reply:
Ok thanks for your review, I will make the changes.